lib/codecs/src/encoding/mod.rs

```diff
@@ -335,8 +350,9 @@ impl Serializer {
     ///
     /// Panics if the serializer does not support encoding to JSON. Call `Serializer::supports_json`
     /// if you need to determine the capability to encode to JSON at runtime.
-    pub fn to_json_value(&self, event: Event) -> Result<serde_json::Value, serde_json::Error> {
+    pub fn to_json_value(&self, event: Event) -> Result<serde_json::Value, vector_core::Error> {
```

Comment:
(In case you missed it), this was kind of discussed with Pablo above in his most recent review, but there may have been undesirable outcomes from changing the fn signature.

Reply:
This has been discussed / suggested in https://github.com/vectordotdev/vector/pull/13333#discussion_r916969968.

Since the `to_json_value` is supposed to encode the value by its format first, it makes sense to expand the error type for more generic errors that can occur either while encoding or converting to `JSON`